src/core/shelf.sol

```diff
@@ -80,54 +80,104 @@ contract Shelf is TitleOwned {
         shelf[loan].tokenId = nft_;
     }
 
-    function file(uint loan, address registry_, uint nft_, uint principal_) public auth {
-        shelf[loan].registry = registry_;
-        shelf[loan].tokenId = nft_;
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    function token(uint loan) public view returns (address registry, uint nft) {
+        return (shelf[loan].registry, shelf[loan].tokenId);
     }
 
-    function file(uint loan, uint principal_) public auth {
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    // --- Shelf: Loan actions ---
+    function issue(address registry, uint token) public returns (uint) {
+        require(NFTLike(registry).ownerOf(token) == msg.sender, "nft-not-owned");
+
+        bytes32 nft = keccak256(abi.encodePacked(registry, token));
+        require(nftlookup[nft] == 0, "nft-in-use");
+
+        uint loan = title.issue(msg.sender);
+        nftlookup[nft] = loan;
+
+        shelf[loan].registry = registry;
+        shelf[loan].tokenId = token;
+
+        return loan;
     }
 
-    // Move the NFT out of the shelf
-    function move(uint loan, address registry_, uint nft_, address to) public owner(loan) {
-        NFTLike(registry_).transferFrom(address(this), to, nft_);
+    function close(uint loan) public owner(loan) {
+        require(pile.debt(loan) == 0, "outstanding-debt"); // TODO: only allow closing of a loan that isn't active anymore. maybe there is a better criteria
+        title.close(loan);
+        bytes32 nft = keccak256(abi.encodePacked(shelf[loan].registry, shelf[loan].tokenId));
+        nftlookup[nft] = 0;
     }
 
-    function release (uint loan, address usr) public owner(loan) {
-        uint debt = pile.debtOf(loan);
-        require(debt == 0, "debt");
-        move(loan, shelf[loan].registry, shelf[loan].tokenId, usr);
-        adjust(loan);
+    // --- Currency actions ---
+    function want() public view returns (int) {
+        return int(balance) - int(tkn.balanceOf(address(this))); // safemath
```

Comment:
This should not return an int but a boolean flag and a value.

Reply:
https://github.com/centrifuge/tinlake/issues/171